### ISSUE TITLE
Viewscreen

### ISSUE THREAD

**Issue by alexanderson1993** [Thorium-Sim/thorium-nova#414]
Discussed in https://github.com/Thorium-Sim/thorium-nova/discussions/41 and https://github.com/Thorium-Sim/thorium-nova/discussions/320

This issue encapsulates all of the features needed for the first iteration of ship's viewscreen. This isn't meant to be all-inclusive, just enough to get started.

- [x] #421
- [x] #415
- [x] #416
- [x] #417
- [x] #420
- [x] #418
- [ ] Viewscreen - Create a system for displaying overlays on the viewscreen.
- [ ] Viewscreen - Create a card for the crew to control what is displayed on the viewscreen.
- [ ] Viewscreen - Create a widget that allows any crew member to view the viewscreen on their computer.

**Comment by 1deejay** [comment]
In part of design for this. One of the frustrating things when participating in a bridge sim is that the Flight Director has full control of the viewscreen. This is addressing that issue certainly, but the controls for what is on the viewscreen should be in full control of the crew.

**Comment by alexanderson1993** [comment]
Yes, crew should be able to affect what's on the view screen. The challenge is that the view screen is one of the primary ways the crew gets information about stuff they should be focusing on. So there's probably going to be a good bit of automation, where the simulation overrides the crew's view screen choices to show information that's crucial to the mission.

This could take many forms, including a notification indicating the crew to choose to adjust their view screen to the view that the simulation wants. 

How all that shakes out is TBD, but you're on the right track. Not necessarily full control, but crew does have control.

**Comment by 1deejay** [comment]
If there were alerts guiding people to the correct thing, that could also help. I'm basing this off of StarCraft. Almost the entire time you have full control of where you are looking at. There are alerts and buttons that pop up that can direct your attention to something and on occasion forces you to look at a specific thing.

With Nova being more crew driven with a flight enhancer for those who use it, having the ability to give maximum control to crew would also help immersion.

**Comment by alexanderson1993** [comment]
Yeah, that makes sense. Kind of like the spacebar (if I'm remembering right) that jumps the camera to the section of the map that caused the last notification. 

A notification on the 1st Officer's station (that's the crew member I would put in charge of the view screen, but it could be anyone) indicating a new contact on the view screen, clicking the notification has the camera track it, and then the crew member can control the camera zoom from their station.

**Comment by 1deejay** [comment]
Thinking through it, I imagine that several stations can send information to the main viewscreen. For example, if there are interesting readings from operations, they can show it onscreen for everyone to see. The pilot can show the outboard cameras so planets can be seen and much more. It becomes a collective space where the captain can direct what they want to see at the moment. This helps fulfill the fantasy of the captain giving the order "on screen" and they can have everyone view what is happening.

This also let's the captain talk to individuals for information that doesn't need the full attention of everyone.

An example of this was when I was last at CMSC. I was first officer and I triaged information for the captain where possible because there is a ton of information coming in all the time. I could speak with different officers at their station and if anything needed a decision from the captain I could bring it up and the captain could have it put on the viewscreen.

Another consideration is that for flights without a director, every station will need access to the viewscreen individually which is something we can set up as a base line.

This is getting into conversations of what the "standard" ship size should be and while it is a blast to play with 8-10 people. A feasible amount for at home play would be 3-4. But that conversation is best held elsewhere and another time, right now we are just getting functionality coded.

**Comment by alexanderson1993** [comment]
> This is getting into conversations of what the "standard" ship size should be.

The intention is that there will be two classes of ships. "Small" ships do not include a ship map and exclude stations like damage teams, security, and medical. These ships would host 1-6 crew members with station configurations similar to other bridge sims, like Artemis, Empty Epsilon, and Starship Horizon. (Captain, Pilot, Weapons, Engineer, Science, and Communications/Operations)

"Big" ships would include a ship map and allow for those extra stations. They would host 5-12 crew members.

The station configuration is dynamically created based on the number of players and the capabilities of the ship when the flight is started.

**Comment by 1deejay** [comment]
I was curious what choices we would go for. Starting with small and large configuration is a good place for sure and most of those questions can't really be answered until we get some prototype and start testing.

The only other thing I want to make clear, since part of the design is for people to play in their homes or even over the internet, it is expected that most players will only have one monitor available to them so having a viewscreen button available to people is certainly a good thing to keep in mind. But we'll leave most of that conversation for when more parts are starting to come together, as I might be wrong altogether.

Edit: It looks like my last paragraph is wholly unnecessary as that was added to the to do list on this issue anyway.
